Cargo.toml

```diff
@@ -28,6 +28,7 @@ toml = "~0.2.1"
 env_logger = "~0.3.4"
 typed-arena = "~1.2.0"
 clap = "~2.19.0"
+parking_lot = "~0.3.6"
```

Comment:
I'm no expert on this, but my impression is that you always have to deal with the possibility of inconsistent data caused by a panicking thread and lock poisoning deals with this by panicking other threads that attempt to access that data. If you don't use poisoning, then you need some other strategy or you'll have rare and subtle bugs.

I don't think the performance of r/w lock will be meaningful in Racer, I would expect it to be dwarfed by the other things that happen. But I haven't profiled or anything, and again I'm not an expert.